django/contrib/admin/static/admin/js/inlines.js

```diff
@@ -218,12 +218,10 @@
             // instantiate a new SelectFilter instance for it.
             if (typeof SelectFilter !== 'undefined') {
                 $('.selectfilter').each(function(index, value) {
-                    const namearr = value.name.split('-');
-                    SelectFilter.init(value.id, namearr[namearr.length - 1], false);
+                    SelectFilter.init(value.id, $(this).data('fieldName'), false);
```

Comment:
Accessing the data attributes doesn't require jQuery. There have been some efforts to reduce jQuery usage in the admin, so can take advantage of that here.

```suggestion
                    SelectFilter.init(value.id, this.dataset.fieldName, false);
```

And the other lines too:

```diff
diff --git a/django/contrib/admin/static/admin/js/inlines.js b/django/contrib/admin/static/admin/js/inlines.js
index 777b76e1fe..d9a9032c45 100644
--- a/django/contrib/admin/static/admin/js/inlines.js
+++ b/django/contrib/admin/static/admin/js/inlines.js
@@ -218,10 +218,10 @@
             // instantiate a new SelectFilter instance for it.
             if (typeof SelectFilter !== 'undefined') {
                 $('.selectfilter').each(function(index, value) {
-                    SelectFilter.init(value.id, $(this).data('fieldName'), false);
+                    SelectFilter.init(value.id, this.dataset.fieldName, false);
                 });
                 $('.selectfilterstacked').each(function(index, value) {
-                    SelectFilter.init(value.id, $(this).data('fieldName'), true);
+                    SelectFilter.init(value.id, this.dataset.fieldName, true);
                 });
             }
         };
@@ -281,10 +281,10 @@
             // If any SelectFilter widgets were added, instantiate a new instance.
             if (typeof SelectFilter !== "undefined") {
                 $(".selectfilter").each(function(index, value) {
-                    SelectFilter.init(value.id, $(this).data('fieldName'), false);
+                    SelectFilter.init(value.id, this.dataset.fieldName, false);
                 });
                 $(".selectfilterstacked").each(function(index, value) {
-                    SelectFilter.init(value.id, $(this).data('fieldName'), true);
+                    SelectFilter.init(value.id, this.dataset.fieldName, true);
                 });
             }
         };
```